### PR TITLE
Add address balance gas payment generation to transaction fuzzer

### PR DESCRIPTION
**Pull request by akmysti**
## Description 

Extend the transaction fuzzer to generate address-balance-style gas
payments (empty payment vector, no gas coin objects) alongside the
existing coin-based payments. This exercises the validation paths for
the newer address balance gas payment feature.

- Add `allow_address_balance` flag to `GasDataGenConfig` with a
  `with_address_balance()` constructor
- Add `generate_address_balance_gas_data()` producing empty-payment
  `GasDataWithObjects`
- Update `Arbitrary` impl to mix coin-based and address-balance
  strategies via `prop_oneof!`
- Add `ValidDuring` variant to `gen_transaction_expiration()` and
  `gen_transaction_expiration_with_bound()` strategies
- Add fuzz tests for address balance gas in normal and dry-run modes

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] gRPC:
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] Indexing Framework:

### CODE REVIEW

**Comment by akmysti** [comment]
> is this ready for review? can you fix the format issues?

yes. fixed formatting.

File: crates/transaction-fuzzer/tests/gas_data_tests.rs [hunk 1]
```diff
@@ -248,21 +249,142 @@ fn test_gas_data_dev_inspect_no_skip_fuzz() {
 }
 
 // ---------------------------------------------------------------------------
-// Deterministic edge case tests
+// Edge case helpers — each returns a GasDataWithObjects for a specific scenario.
+// Used by both deterministic tests (for error assertions) and proptests (for
+// panic-freedom coverage via prop_oneof!).
+// ---------------------------------------------------------------------------
+
+fn make_edge_case_valid(rgp: u64) -> GasDataWithObjects {
```
Comment (dariorussi) [inline crates/transaction-fuzzer/tests/gas_data_tests.rs]: I am not sure about the edge_case in the name, it feels strange.
Should we just remove it and make them more obvious? e.g. make_gas_data, make_gas_data_price_zero or something similar?
